gdal/swig/include/cpl.i

```diff
@@ -520,6 +521,27 @@ void wrapper_VSIFileFromMemBuffer( const char* utf8_path, int nBytes, const GByt
 #endif
 #endif
 
+#if defined(SWIGJAVA)
+%inline %{
+void wrapper_VSIFileFromMemBuffer( const char* utf8_path, void* nioBuffer, long nioBufferSize) {
+    VSIFCloseL(VSIFileFromMemBuffer(utf8_path, (GByte*) nioBuffer, nioBufferSize, FALSE));
+}
+%}
+#endif
+
+#if defined(SWIGJAVA)
+%apply (jobject outBuffer) {jobject};
+%inline %{
+jobject wrapper_VSIGetMemFileBuffer(JNIEnv *jenv, const char* utf8_path) {
+    vsi_l_offset outBufferLen = 0;
+    GByte* result = VSIGetMemFileBuffer(utf8_path, &outBufferLen, FALSE);
+    return jenv->NewDirectByteBuffer((void*) result, outBufferLen);
```

Comment:
NewDirectByteBuffer() can only work if it is guaranteed that result won't change while the ByteBuffer is in use. There is no such guarantee the /vsimem/ file could be destroyed in-between, or its size grown, causing a realloc(), which would invalidate the original buffer. A copy should be done for safety.
The test program to add mentioned above should also test that.